mrmustard/lab_dev/circuit_components.py

```diff
@@ -110,6 +112,22 @@ def __init__(
                 if self._representation:
                     self._representation = self._representation.reorder(tuple(perm))
 
+    @property
+    def _index_representation(self):
+        r"""
+        A hidden property that keeps track of representations for each index. If it does not exist already,
+        we initialize to Bargmann (B) on all indices.
+        """
+        if not hasattr(self, "_index_representation_"):
+            self._index_representation_ = {i: ("B", None) for i in self.wires.indices}
+        return self._index_representation_
+
+    @_index_representation.setter
+    def _index_representation(self, value):
+
+        # add some validation
+        self._index_representation_ = value
+
     def _serialize(self) -> tuple[dict[str, Any], dict[str, ArrayLike]]:
```

Comment:
Hmm.. not sure about that. But I guess it makes some sense to keep `_index_representation`as a hidden attribute since there are many calls to circuit component initializer and I might be missing correcting some when doing the change.